src/models/searchservice/response/Result.ts

```diff
@@ -10,7 +10,7 @@ export interface Result {
   /** Raw entity profile data in the shape of key-value pairs. */
   rawData: Record<string, unknown>,
   /** {@inheritDoc Source}*/
-  source: Source,
+  source: Source | string,
```

Comment:
Could we restrict `source` to the specific strings in the enum instead of any string?

Reply:
Looks like backend may return custom strings that's outside of the defined set of enums we currently have. I will confirm with watson if it can be any string or if there's other defined strings we didn't account for

Reply:
Good call! Confirmed with Watson that it will always be one of the 6 enum values specified in this pr's description. So we only need to update GENERIC to CUSTOM_SEARCHER enum.